### PR TITLE
Expose OOR send idempotency keys

### PR DESCRIPTION
**Pull request by bhandras**
## Summary

This PR replaces the local OOR session-listing recovery surface with a smaller, caller-driven idempotency key for OOR sends. The new surface lets a caller attach a stable key to a send intent and lets the client carry that key through the daemon RPC, SDK, durable actor messages, restored session handles, outgoing snapshots, and state transitions.

The branch is split into four commits:

- remove the unused local OOR session listing RPC, CLI command, SDK helpers, actor summary plumbing, and tests added in #316
- add durable/RPC/SDK support for caller-provided OOR send idempotency keys
- make keyed outgoing OOR starts idempotent in the actor and have the daemon probe that keyed state before doing any send construction work
- expose the key on `darepocli send oor`, the MCP `send_oor` tool, and the public command schema

## Why

The existing actor replay check only deduplicates byte-equivalent start requests. It rebuilds the OOR package, derives the Ark txid/session ID, and then checks whether that session already exists.

That is enough for exact replays, but not for retries of the same higher-level send intent that are reconstructed differently. For example, after a crash or restart, coin selection can pick a different input for the same recipient amount. That produces a different Ark txid, so the old session-ID check treats it as a new transfer and can emit a second submit package.

A second local review found one more gap in the first version of this branch: the daemon was still selecting and locking wallet VTXOs before reaching the actor's idempotency lookup. That meant a same-key retry could fail at wallet selection when the original send had already reserved all available funds, instead of returning the locally known session. The same ordering issue could also mask existing keyed sessions behind custom-input parsing or reservation failures.

A caller-provided idempotency key gives the actor a stable intent identifier that is independent of selected inputs and generated txids. Once a keyed outgoing session is locally known, the daemon asks the actor for that key before recipient derivation, wallet selection, custom input reservation, transfer input construction, change creation, or package submission.

## Behavior

Empty idempotency keys preserve the historical behavior.

If a keyed OOR session already exists locally, `SendOOR` returns the existing session ID before constructing another transfer. This covers retries where new input selection would otherwise produce a different Ark txid, and it also covers the no-spare-funds case where input selection would otherwise fail before the retry could reach the actor.

The actor response still reports whether a start request reused an existing session after construction. The daemon RPC uses that signal to unlock any wallet-selected VTXOs from deterministic duplicate requests, because those freshly selected inputs were not consumed by the original session. This prevents duplicate starts from progressively leaving unrelated wallet funds reserved.

The key is included in snapshots and restored session handles. After a durable checkpoint restore, a retry with the same key can still find the original outgoing session instead of creating a second one.

If a caller crashes before the daemon accepts and persists the actor start message, there is still no local session to find. A retry with the same key creates the first durable session normally.

## Tests

The actor and daemon coverage now prove the relevant retry shapes:

- without a key, the same recipient amount with a different selected input creates a different session ID and emits two server messages
- with a key, the same changed-input retry returns the original session ID and emits only one server message
- after a durable checkpoint restore, the same keyed changed-input retry still returns the restored session and emits no new server message
- through the daemon RPC path, a same-key retry returns the original session before wallet selection when no additional selectable wallet VTXOs are available
- through the daemon RPC path, a deterministic duplicate start unlocks the newly selected wallet VTXO that the actor did not consume
- the public command schema advertises the OOR idempotency key

Local checks run:

- `go test ./oor ./darepod ./sdk/ark ./cmd/darepocli/darepoclicommands`
- `make lint`
- `make commitmsg-lint range=origin/main..HEAD`

### CODE REVIEW

File: oor/actor.go [hunk 1]
```diff
@@ -1814,6 +1825,9 @@ func summaryForSessionHandle(sessionID SessionID,
 		if err != nil {
 			return SessionSummary{}, err
 		}
+		if snapshot.IdempotencyKey == "" {
+			snapshot.IdempotencyKey = handle.IdempotencyKey
+		}
```
[the remark this answers is not in the document]
Comment (bhandras) [reply oor/actor.go]: Addressed by rewriting this branch: the extra local OOR listing/summary surface from #316 was removed, and the remaining idempotency-key snapshot path no longer has the redundant fallback this outdated diff was pointing at.

File: oor/actor.go [hunk 2]
```diff
@@ -2006,7 +2031,15 @@ func (h *sessionHandle) clearRetryMetadata() {
 
 // applyRetrySnapshot copies retry metadata onto an exported snapshot.
```
[the remark this answers is not in the document]
Comment (bhandras) [reply oor/actor.go]: Addressed. The comment now describes that `applyRetrySnapshot` copies both retry metadata and the idempotency key.

**Comment by bhandras** [comment]
@claude please review

**Comment by bhandras** [comment]
@claude pls re-review

**Comment by bhandras** [comment]
Addressed the remaining review notes:

- added the `stateIdempotencyKey` default-branch comment to make it explicit that `Idle` and incoming states intentionally carry no outgoing caller intent key
- added actor-level durable restart coverage for idempotency keys: start a keyed send, stop the actor after the checkpointed submit state, recreate it with the same actor ID/store, retry the same key with a different selected input, and assert we return the restored session without emitting a new server message
- updated the PR description to call out the restart behavior and test coverage

Local checks are green:

- `go test ./oor ./darepod ./sdk/ark ./cmd/darepocli/darepoclicommands`
- `make lint`
- `make commitmsg-lint range=origin/main..HEAD`

@claude pls re-review

**Comment by bhandras** [comment]
Addressed the remaining style gap from pass 3:

- added doc comments for the three new actor idempotency tests, each starting with the test name per the local style rule
- folded the comments into the actor idempotency commit

Local checks are green again:

- `go test ./oor ./darepod ./sdk/ark ./cmd/darepocli/darepoclicommands`
- `make lint`
- `make commitmsg-lint range=origin/main..HEAD`

@claude pls re-review

**Comment by bhandras** [comment]
Addressed the local review findings:

- fixed the daemon RPC wallet-lock leak on same-key retries by having `StartTransferResponse` report when the actor returned an existing session; `SendOOR` now releases any freshly selected wallet VTXOs on that existing-session success path, and also cleans them up on the unexpected-response error path
- added daemon-level regression coverage that drives two `SendOOR` RPC calls with the same idempotency key but different wallet-selected VTXOs, then asserts the second unused selection is unlocked
- wired `idempotency_key` through the MCP `send_oor` tool and added it to the public `send.oor` schema coverage
- updated the PR description and commit bodies to describe these edge cases

Local checks are green:

- `go test ./oor ./darepod ./sdk/ark ./cmd/darepocli/darepoclicommands`
- `make lint`
- `make commitmsg-lint range=origin/main..HEAD`

@claude pls re-review

**Comment by bhandras** [comment]
@claude please re-review. We found one additional issue locally after your last pass: same-key SendOOR retries still went through daemon wallet selection before the actor idempotency lookup, so a retry could fail with no spare live VTXOs after the original send reserved the available funds instead of returning the existing session.\n\nI folded a fix into the idempotent-send commit: the daemon now probes the OOR actor by idempotency key before recipient derivation, wallet selection, custom input reservation, transfer input construction, change building, or package submission. I also added daemon coverage where the fake wallet only has one selectable VTXO; the second same-key retry must return the original session without another wallet selection. The deterministic duplicate path still keeps the Existing response/unlock coverage for selected-but-unused wallet inputs.\n\nLocal checks are green: `go test ./oor ./darepod ./sdk/ark ./cmd/darepocli/darepoclicommands`, `make lint`, and `make commitmsg-lint range=origin/main..HEAD`.
